src/commands/open.ts

```diff
@@ -0,0 +1,186 @@
+import chalk from "chalk";
+import open from "open";
+import { getModels } from "../services/app/app.js";
+import { ArgError } from "../services/command/arg.js";
+import type { Command, Usage } from "../services/command/command.js";
+import { UnknownDirectoryError } from "../services/filesync/error.js";
+import { SyncJson, loadSyncJsonDirectory } from "../services/filesync/sync-json.js";
+import { select } from "../services/output/prompt.js";
+import { sprint, sprintln2, sprintlns2 } from "../services/output/sprint.js";
+import { sortBySimilar } from "../services/util/collection.js";
+import { args as PushArgs } from "./push.js";
+
+export const usage: Usage = (ctx) => {
+  if (ctx.args["-h"]) {
+    return sprint`
+    Open a specified Gadget location directly in your browser.
+
+    {bold USAGE}
+
+      ggt open [SUBCOMMAND] [--app=<name>] [--environment=<env>]
+                 [--show-all]
+    
+    {bold EXAMPLES}
+      $ ggt open logs
+      $ ggt open permissions 
+      $ ggt open data modelA
+      $ ggt open data --show-all
+      $ ggt open schema modelA
+      $ ggt open schema --show-all
+    
+    {bold FLAGS}
+      -a, --app=<name>      The application to open
+      -e, --from=<env>      The environment to open
+          --show-all        Shows all available models to open
+    
+    Run "ggt open --help" for more information.
+    `;
+  }
+
+  return sprint`
+    Open a specified Gadget location directly in your browser.
+
+    {bold USAGE}
+
+      ggt open [SUBCOMMAND] [--app=<name>] [--environment=<env>]
+                 [--show-all]
+
+    {bold SUBCOMMANDS}
+      • logs
+        The log viewer for the app on the current environment
+      • permissions
+        The permissions settings for the app on the current environment
+      • data
+        The data viewer for a specified model on the current environment
+      • schema
+        The schema viewer for a specified model on the current environment
+    
+    {bold EXAMPLES}
+
+      $ ggt open logs
+      $ ggt open permissions 
+      $ ggt open data modelA
+      $ ggt open data --show-all
+      $ ggt open schema modelA
+      $ ggt open schema --show-all
+
+    {bold FLAGS}
+
+      -a, --app, --application=<name>
+        The application to open.
+
+        Defaults to the application within the ".gadget/sync.json"
+        file in the current directory or any parent directories.
+
+      -e, --from, --env, --environment=<name>
+        The development environment to open.
+
+        Defaults to the environment within the ".gadget/sync.json"
+        file in the current directory or any parent directories.
+
+      --show-all, 
+        Shows a list of available models the user may select from to open
+`;
+};
+
+export type OpenArgs = typeof args;
+
+export const args = {
+  ...PushArgs,
```

Comment:
Let's spread `SyncJsonArgs` rather than `PushArgs` since `--force` doesn't make sense for `ggt open`.

https://github.com/gadget-inc/ggt/blob/main/src/services/filesync/sync-json.ts#L20-L25